pkg/validate/security_context_linux.go

```diff
@@ -639,6 +589,151 @@ var _ = framework.KubeDescribe("Security Context", func() {
 		})
 	})
 
+	Context("SupplementalGroupsPolicy", func() {
+		BeforeEach(func(ctx context.Context) {
+			By("skip if the runtime does not support SupplementalGroupsPolicy")
+			statusResponse, err := rc.Status(ctx, false)
+			Expect(err).NotTo(HaveOccurred())
+			if !(statusResponse.Features != nil && statusResponse.Features.SupplementalGroupsPolicy) {
+				Skip("The runtime does not support SupplementalGroupsPolicy feature")
+			}
+		})
+
+		When("SupplementalGroupsPolicy=Merge (Default)", func() {
+			It("if the container's primary UID belongs to some groups in the image, runtime should add SupplementalGroups to them", func() {
+				By("create pod")
+				podID, podConfig, podLogDir = createPodSandboxWithLogDirectory(rc)
+
+				By("create container for security context SupplementalGroups")
+				supplementalGroup := int64(1234)
+				containerName := "container-with-SupplementalGroupsPolicyMerge-" + framework.NewUUID()
+				logPath := containerName + ".log"
+				containerConfig := &runtimeapi.ContainerConfig{
+					Metadata: framework.BuildContainerMetadata(containerName, framework.DefaultAttempt),
+					Image:    &runtimeapi.ImageSpec{Image: testImagePreDefinedGroup},
+					Command:  []string{"sh", "-c", "id -G; while :; do sleep 1; done"},
+					Linux: &runtimeapi.LinuxContainerConfig{
+						SecurityContext: &runtimeapi.LinuxContainerSecurityContext{
+							RunAsUser:          &runtimeapi.Int64Value{Value: imagePredefinedGroupUID},
+							SupplementalGroups: []int64{supplementalGroup},
+							// SupplementalGroupsPolicy_Merge is default(0)
+							// SupplementalGroupsPolicy: runtimeapi.SupplementalGroupsPolicy_Merge,
```

Comment:
Do you want to keep this comment here?

Reply:
fixed in https://github.com/kubernetes-sigs/cri-tools/pull/1438/commits/c45cf828ce27fc728c7e3342d87cd716d295be04